client/src/routes/+layout.server.ts

```diff
@@ -1,20 +1,70 @@
 import { SERVER_URL } from "$env/static/private";
-import favorites from "$lib/data";
 import api from "$lib/server/api";
+import { z } from "zod";
 import type { LayoutServerLoad } from "./$types";
 
-export const load = (async () => {
-    const go_server_favorites = await api(SERVER_URL + "/favorites");
-    if (!go_server_favorites.success) {
-        console.error(
-            "Failed to load favorites from server",
-            go_server_favorites.error,
+export type Favorite = {
+    id: string; 
+    anime_id: string;
+    anime_title: string;
+    anime_image_url: string;
+    created: string;
+    updated: string;
+};
+
+const favoriteSchema = z.object({
+    id: z.string(),
+    anime_id: z.string(),
+    anime_title: z.string(),
+    anime_image_url: z.string(),
+    created: z.string().refine(date => !isNaN(Date.parse(date)), {
+        message: "Invalid date format"
+    }),
+    updated: z.string().refine(date => !isNaN(Date.parse(date)), {
+        message: "Invalid date format"
+    }),
+});
+
+
+export const load: LayoutServerLoad = (async () => {
+    try {
+        const response = await api(`${SERVER_URL}/favorites`);
+
+        if (!response.success) {
+            console.error(
+                "Failed to load favorites from server",
+                response.error
+            );
+            return {
+                favorites: new Map() 
+            };
```

Comment:
svelte have special function to handle errors, that will take user to the closest +error.svelte page.

```
throw error(500, "Error msg");
```